software/apt-lib.pl

```diff
@@ -457,7 +457,34 @@ sub list_package_repos
 foreach my $f ($sources_list_file, glob("$sources_list_dir/*")) {
 	my $lref = &read_file_lines($f, 1);
 	my $lnum = 0;
+	my (%repo, @types);
+	my $repo = sub {
```

Comment:
Why use an anonymous sub here? It seems like this is only called once

Reply:
It won't fit within the 80-character-per-line limit because there are too many tabs and code.

Reply:
Ok but then why is it defined inside the loop?

Reply:
> but then why is it defined inside the loop?

This is because it uses locally scoped variables that are only accessible within the current context of the specific repo file.